spec/fortnox/api/base_spec.rb

```diff
@@ -96,15 +96,15 @@
       )
     end
 
-    it 'works', focus: true do
+    it 'works' do
       api = Fortnox::API::Base.new
       response1 = api.get( '/test', { body: '' })
       response2 = api.get( '/test', { body: '' })
       response3 = api.get( '/test', { body: '' })
 
-      expect( response1.parsed_response ).to_not eq( response2.parsed_response )
-      expect( response3.parsed_response ).to_not eq( response2.parsed_response )
-      expect( response1.parsed_response ).to eq( response3.parsed_response )
+      expect( response1 ).to_not eq( response2 )
+      expect( response3 ).to_not eq( response2 )
```

Comment:
As you wrote above:
> since the return is in a block it will return not only the block but the entire function the block is in...
The entire function was returned before. That's why you had to run `parsed_response`. Now, when the entire function is no longer returned but only the actual response this is the correct implementation.